chained/water_version_control.go

```diff
@@ -0,0 +1,274 @@
+package chained
+
+import (
+	"context"
+	"io"
+	"os"
+	"path/filepath"
+	"slices"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/getlantern/flashlight/v7/proxied"
+	"github.com/gocarina/gocsv"
+)
+
+type versionControl struct {
+	// contain all wasm files available locally
+	// key is something like: protocol.version
+	wasmFilesAvailable map[string]wasmInfo
+	dir                string
+
+	// history of the last time the WASM file was loaded
+	history      []history
+	historyMutex *sync.Mutex
+}
+
+type history struct {
+	Transport      string    `csv:"transport"`
+	LastTimeLoaded time.Time `csv:"last_time_loaded"`
+}
+
+type wasmInfo struct {
+	version   string
+	protocol  string
+	builtWith string
```

Comment:
It doesn't look like these fields are actually used anywhere.  All we're actually doing is naming the files based on whatever the back-end tells us (and insisting these names contain two periods), then storing the WASM in those files.  Let's just keep it simple and not try to interpret the names at all.